dll/win32/kernel32/client/fiber.c

```diff
@@ -312,32 +312,88 @@ IsThreadAFiber(VOID)
 }
 
 /*
- * @unimplemented
+ * @implemented
  */
 DWORD
 WINAPI
 FlsAlloc(PFLS_CALLBACK_FUNCTION lpCallback)
 {
-   (void)lpCallback;
+    DWORD dwFlsIndex;
+    PPEB Peb = NtCurrentPeb();
+    PVOID *ppFlsSlots;
+
+    RtlAcquirePebLock();
 
-   UNIMPLEMENTED;
-   SetLastError(ERROR_CALL_NOT_IMPLEMENTED);
-   return FLS_OUT_OF_INDEXES;
+    ppFlsSlots = NtCurrentTeb()->FlsData;
+
+    if (!Peb->FlsCallback &&
+        !(Peb->FlsCallback = RtlAllocateHeap(RtlGetProcessHeap(), HEAP_ZERO_MEMORY,
+                                             FLS_MAXIMUM_AVAILABLE * sizeof(PVOID))))
+    {
+        SetLastError(ERROR_NOT_ENOUGH_MEMORY);
```

Comment:
Please change all the "SetLastError" in the FLS functions to "BaseSetLastNTError" , as is done in the rest of kernel32 code (see also the TLS functions).

Reply:
There is also code in kernel32 using `SetLastError`